test_runner/fixtures/neon_fixtures.py

```diff
@@ -2343,22 +2343,49 @@ def assert_no_metric_errors(self):
             value = self.http_client().get_metric_value(metric)
             assert value == 0, f"Nonzero {metric} == {value}"
 
-    def log_contains(self, pattern: str) -> Optional[str]:
-        """Check that the pageserver log contains a line that matches the given regex"""
+    def log_contains(
+        self, pattern: str, skip_until: "None | str | Callable[[str], bool]"
+    ) -> Optional[str]:
+        """
+        Check that the pageserver log contains a line that matches the given regex.
+        Use `skip_until` to limit the search to the suffix of the log that follows the first line that matches `skip_until`.
+        """
         logfile = self.workdir / "pageserver.log"
         if not logfile.exists():
             log.warning(f"Skipping log check: {logfile} does not exist")
             return None
 
         contains_re = re.compile(pattern)
 
+        skip_until_finder: Callable[[str], bool]
+        if skip_until is None:
+
+            def always_true_finder(_line: str) -> bool:
+                return True
+
+            skip_until_finder = always_true_finder
+        elif isinstance(skip_until, str):
+            skip_until_pattern_re = re.compile(skip_until)
```

Comment:
```suggestion
            skip_until_pattern_re = re.compile(re.escape(skip_until))
```

Reply:
Why this? We don't do it for the `pattern` either.